### PR TITLE
Add sync command

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/14116

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by corest** [comment]
I'm not super sure if it is okay to mix flags in this way between two commands because I didn't see operators with additional commands before. 

Maybe it makes sense to move `kubernetes.*` flags to the root command, so that instead of 

```
silence-operator daemon --kubernetes.incluster ...
```

use 

```
silence-operator --kubernetes.incluster daemon/sync
```

So that we reuse flags across commands

File: cmd/sync/runner.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+				return microerror.Mask(err)
+			}
+			silenceFiles = append(silenceFiles, files...)
+		}
+	}
+
+	tags := make(map[string]string)
+	{
+		for _, tag := range r.flag.Tags {
+			tagObj := strings.Split(tag, "=")
```
Comment (kopiczko) [inline cmd/sync/runner.go:64]:
```suggestion
			tagObj := strings.SplitN(tag, "=", 1)
```

File: cmd/sync/runner.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+			tagValue := ""
+			if len(tagObj) == 2 {
+				tagValue = tagObj[1]
+			}
+
+			tags[tagName] = tagValue
+		}
+	}
+
+	var desiredSilences []monitoringv1alpha1.Silence
```
Comment (kopiczko) [inline cmd/sync/runner.go:75]: IMO they shoudl be called filteredSilences

File: cmd/sync/runner.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	{
+		for _, silenceFile := range silenceFiles {
+			data, err := ioutil.ReadFile(silenceFile)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			var desiredSilence monitoringv1alpha1.Silence
```
Comment (kopiczko) [inline cmd/sync/runner.go:83]:
```suggestion
			var silence monitoringv1alpha1.Silence
```

File: cmd/sync/command.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	"os"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/cobra"
+)
+
+const (
+	name        = "sync"
+	description = "Synchronize CRs from local directory into Kubernetes API."
```
[the remark this answers is not in the document]
Comment (corest) [reply cmd/sync/command.go:14]: added readme

**Comment by corest** [comment]
Addressed comments
PTAL @kopiczko

File: README.md [hunk 5]
```diff
@@ -3,28 +3,92 @@
...
+    value: test
+    isRegex: false
+```
+
+- `targetTags` field defines a list of tags, which `sync` command uses to match CRs towards a specific environment.
+
+For example, to ensure raw CR, stored at `/folder/cr.yaml`, run:
+
+```bash
+silence-operator sync --tag installation=kind --tag provider=local --dir /folder`
```
Comment (kopiczko) [inline README.md:47]:
```suggestion
silence-operator sync --tag installation=kind --tag provider=local --dir /folder
```

File: cmd/sync/command.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (kopiczko) [reply cmd/sync/command.go:14]: Why not dropping `silence-operator sync --tag installation=kind --tag provider=local --dir /folder` here as well? So it's visible when doing `silence-operator sync -h`

File: README.md [hunk 6]
```diff
@@ -3,28 +3,92 @@
...
+
+## Overview
+
+### CustomResourceDefinition
+
+The silence-operator monitors the Kubernetes API server for changes
+to `Silence` objects and ensures that the current Alertmanager alerts match these objects.
+The Operator acts on the following [custom resource definitions (CRDs)](https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/):
+
+`Silence` CRD definition can be found [here](https://github.com/giantswarm/apiextensions/blob/master/pkg/apis/monitoring/v1alpha1/silence_types.go).
```
Comment (kopiczko) [inline README.md:16]:
```suggestion
The Operator reconciles the `Silence` [Custom Resource Definition (CRD)][crd] which
can be found [here][silence-crd].

[crd]: https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/
[silence-crd]: https://github.com/giantswarm/apiextensions/blob/master/pkg/apis/monitoring/v1alpha1/silence_types.go
```

File: README.md [hunk 7]
```diff
@@ -3,28 +3,92 @@
...
+Build the standard way.
+
+```
+go build github.com/giantswarm/silence-operator
+```
+
+## Contact
+
+- Mailing list: [giantswarm](https://groups.google.com/forum/!forum/giantswarm)
+- IRC: #[giantswarm](irc://irc.freenode.org:6667/#giantswarm) on freenode.org
```
Comment (kopiczko) [inline README.md:75]:
```suggestion
```

I think we decided to drop IRC support a while back.

**Review by kopiczko — APPROVED** [review]

**Pull request merged**